ads/google/utils.js

```diff
@@ -133,3 +134,32 @@ function validateDimensions(width, height, widthCond, heightCond, errorBuilder)
   }
   return !badParams.length;
 }
+
+const WHITELISTED_HEIGHT = 320;
+
+/**
+ * Check that the <amp-ad> tag height is an allowed heights for responsive ad
+ * units.
+ * @param {number} height
+ */
+function validateResponsiveAdHeight(height) {
+  user().assert(
+      height === WHITELISTED_HEIGHT,
+      'Specified height ' + height +
+          ' in <amp-ad> tag is not equal to the required height of ' +
+          WHITELISTED_HEIGHT + ' for responsive AdSense ad units.');
+}
+
+/**
+ * Validate <amp-ad> data, and validate height for responsive ad units.
+ */
+export function validateAdSenseData(data, height) {
+  // TODO: check mandatory fields
+  validateData(data, [], [
+    'adClient', 'adSlot', 'adHost', 'adtest', 'tagOrigin', 'experimentId',
+    'ampSlotIndex', 'adChannel', 'autoFormat', 'a4aUpgradeType', 'fullWidth',
```

Comment:
Why did you add a4aUpgradeType?  That element is actually generated as part of the fast fetch flow for debugging purposes and so you can remove.